ludwig/schema/features/text_feature.py

```diff
@@ -0,0 +1,65 @@
+from typing import Optional
+
+from marshmallow_dataclass import dataclass
+
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.preprocessing import BasePreprocessingConfig, PreprocessingDataclassField
+
+
+@dataclass
+class TextInputFeatureConfig(schema_utils.BaseMarshmallowConfig):
+    """TextInputFeatureConfig is a dataclass that configures the parameters used for a text input feature."""
+
+    preprocessing: BasePreprocessingConfig = PreprocessingDataclassField(feature_type="text")
+
+    encoder: Optional[str] = schema_utils.StringOptions(
```

Comment:
Seeing a list this long made me think: is there a way to pull this list from code? I think one thing I get a little concerned about is that it will be hard to maintain this list and keep it in sync with the code.

Reply:
@connor-mccorm I wonder if you can use `encoder_registry` here? (tho this is just from a quick glance)

Reply:
You can use `list(encoder_registry["text"].keys())`